From c631eac735fba1dd0993d8219f048cc775136d3d Mon Sep 17 00:00:00 2001
From: Mike Lothian <mike@fireburn.co.uk>
Date: Sat, 12 Mar 2022 01:40:01 +0000
Subject: [PATCH 2/2] Revert "vulkan/wsi: handle queue families properly for
 non-concurrent sharing mode."

This reverts commit 48b3ef625e1909c1be31fbe10adb53734af38eb4.
---
 src/vulkan/wsi/wsi_common.c | 24 +++++++++---------------
 1 file changed, 9 insertions(+), 15 deletions(-)

diff --git a/src/vulkan/wsi/wsi_common.c b/src/vulkan/wsi/wsi_common.c
index 5c89e7ceb1f..5351b83a4f3 100644
--- a/src/vulkan/wsi/wsi_common.c
+++ b/src/vulkan/wsi/wsi_common.c
@@ -362,22 +362,16 @@ wsi_configure_image(const struct wsi_swapchain *chain,
                     struct wsi_image_info *info)
 {
    memset(info, 0, sizeof(*info));
-   uint32_t *queue_family_indices;
 
-   if (pCreateInfo->imageSharingMode == VK_SHARING_MODE_CONCURRENT) {
-      queue_family_indices =
-         vk_alloc(&chain->alloc,
-                  sizeof(*queue_family_indices) *
-                  pCreateInfo->queueFamilyIndexCount,
-                  8, VK_SYSTEM_ALLOCATION_SCOPE_OBJECT);
-      if (!queue_family_indices)
-         goto err_oom;
-
-      for (uint32_t i = 0; i < pCreateInfo->queueFamilyIndexCount; i++)
-         queue_family_indices[i] = pCreateInfo->pQueueFamilyIndices[i];
-   } else {
-      queue_family_indices = NULL;
-   }
+   uint32_t *queue_family_indices =
+      vk_alloc(&chain->alloc,
+               sizeof(*queue_family_indices) *
+               pCreateInfo->queueFamilyIndexCount,
+               8, VK_SYSTEM_ALLOCATION_SCOPE_OBJECT);
+   if (!queue_family_indices)
+      goto err_oom;
+   for (uint32_t i = 0; i < pCreateInfo->queueFamilyIndexCount; i++)
+      queue_family_indices[i] = pCreateInfo->pQueueFamilyIndices[i];
 
    info->create = (VkImageCreateInfo) {
       .sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO,
-- 
2.35.1

